har2requests/__init__.py

```diff
@@ -188,6 +210,43 @@ def main(src, unsafe, no_infer, hide_result, include_options):
             header_to_variable=header_to_variable,
             file=wrapper,
         )
+        if generate_assertions:
+            output(
+                f"assert r.status_code == {request.responseStatus}, f'Expected status {request.responseStatus} but was {{r.status_code}} for url \"{request.url}\"'"
+            )
+
+        if debug_requests:
+            output('# request headers:')
+            output("#"
+                + json.dumps(request.headers, indent=2)
+                .strip()
+                .replace("\n", "\n# "))
+            output('# request cookies:')
+            output("#"
+                + json.dumps(request.cookies, indent=2)
+                .strip()
+                .replace("\n", "\n# "))
+            output('# response headers:')
+            output("#"
+                + json.dumps(request.responseHeaders, indent=2)
+                .strip()
+                .replace("\n", "\n# "))
+            output('# response cookies:')
+            output("#"
+                + json.dumps(request.responseCookies, indent=2)
+                .strip()
+                .replace("\n", "\n# "))
+
+        if generate_assertions:
+            expected_cookie_names = set(list(k for k,v in request.cookies.items() if v != '') + list(request.responseCookies.keys()))
+            for k in expected_cookie_names.copy():
+                if k in request.responseCookies and request.responseCookies[k] == '':
+                    expected_cookie_names.remove(k)
+            expected_cookie_names_safe = str(list(map(lambda str: re.sub("'", "\\'", str), expected_cookie_names)))
+            output(
+                f"assert set({expected_cookie_names_safe}) == set(s.cookies.get_dict().keys()), f\"Expected defined cookies to be {expected_cookie_names_safe} but was {{s.cookies.get_dict().keys()}} for url \\\"{request.url}\\\"\""
```

Comment:
wow that's some quote-fu

Reply:
Ouch. Yes it is a bit over the top. Yuck.